### DISCUSSION TITLE
Latex has failed

### DISCUSSION THREAD

**Discussion by eduardomartini** [allefeld/pytikz#6] [Q&A] [answered]
Hi,

I just found out about this project, and it is exactly was I what looking for to get some figures for a paper I am witting, so thanks s a lot for sharing!

Just one question. After installing the package, I can easily import the package in python and add stuff to the plot. But when I try any simulation method (demo, write_image, etc), I get a "tikz.LatexError: LaTeX has failed-".

Any suggestions what may be going on? Is there any extra setup to get latex machinery running under the hood?

Thanks again for sharing the project,

Best,
Eduardo

**Comment by allefeld** [comment, marked as answer]
It's hard to tell from your description. Is there any additional message after "LaTeX has failed"?

A possibility is: The module uses by default XeLaTeX – maybe you don't have that installed? If not, you can either install it, or do
```python
from tikz import cfg
cfg.latex = 'pdflatex'
```
or some other value, it should be the name of the LaTeX binary.

Or you could get the code for the picture from `pic.document_code()`, write it to a file, try to compile yourself, and see if and which error comes up.

**Reply by eduardomartini** [reply]
Thanks a lot for the reply!

I do have xelatex installed. Although I use it rarely, but some I write some documents with it. Nevertheless, switching to pfdlatex as you suggested did the trick, so I believe it really is just some missing configuration on my side.

Thanks a lot for the help!
Eduardo
